state/execution.go

```diff
@@ -86,7 +86,7 @@ func (blockExec *BlockExecutor) ApplyBlock(state State, blockID types.BlockID, b
 	fail.Fail() // XXX
 
 	// update the state with the block and responses
-	state, err = updateState(state, blockID, block.Header, abciResponses)
+	state, err = updateState(state, blockID, &block.Header, abciResponses)
```

Comment:
this one needs a pointer?

Reply:
I did not want to remove all the pointers (can result in poor performance, no?)